python/cog/predictor.py

```diff
@@ -326,6 +332,80 @@ class Output(BaseModel):
     return OutputType
 
 
+def get_train(predictor: Any) -> Callable:
+    if hasattr(predictor, "train"):
+        return predictor.train
+    return predictor
+
+
+def get_training_input_type(predictor: BasePredictor) -> Type[BaseInput]:
+    """
+    Creates a Pydantic Input model from the arguments of a Predictor's train() method.
+
+    def train(self, text: str):
+        ...
+
+    programmatically creates a model like this:
+
+    class TrainingInput(BaseModel):
+        text: str
+    """
+
+    train = get_train(predictor)
+    signature = inspect.signature(train)
+
+    return create_model(
+        "TrainingInput",
+        __config__=None,
+        __base__=BaseInput,
+        __module__=__name__,
+        __validators__=None,
+        **get_input_create_model_kwargs(signature),
+    )  # type: ignore
+
+
+def get_training_output_type(predictor: BasePredictor) -> Type[BaseModel]:
+    """
+    Creates a Pydantic Output model from the return type annotation of a train() method.
+    """
+
+    # class TrainingOutput(BaseModel):
+    #     weights: CogFile
+
+    # return TrainingOutput
+
+    train = get_train(predictor)
+    signature = inspect.signature(train)
+
+    # raise NotImplementedError(f"{signature.return_annotation}")
+
```

Comment:
should this also check that this has a `weights` key? Currently omitting that causes image creation to fail in `api`